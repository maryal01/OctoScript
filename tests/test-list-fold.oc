list<int> nums = [10, 11, 12, 13, 14, 15];

lambda<int, int => int> add = fn (int acc, int elem) => {acc + elem};
int sum = fold_left(add, 0, nums);

lambda<int, int => int> add_even = fn (int acc, int elem) => {if (elem % 2 == 0) ? elem + acc : acc};
int sum_even = fold_left(add_even, 0, nums);

print("list sum: %d, even sum: %d\n", sum, sum_even);

list<string> words = ["one", "two", "three", "four", "five"];
int total_chars = fold_left(fn(int acc, string word) => {acc + strlen(word)}, 0, words);
print("total characters: %d", total_chars);