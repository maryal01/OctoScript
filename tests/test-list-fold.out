list sum: 75, even sum: 36
total characters: 19